### PR TITLE
Fixes #2810 Option to Remove a new language description while uploading

### LINKED ISSUE #2810: Option to Remove a new language description while uploading
**Summary:** 

Sometimes, a user might accidentally hit the "+" when uploading a photo, there is no option to remove it. We should add a "-" so that the user can remove it.

**Steps to reproduce:** 
Upload a new photo, hit the + button many times in Step 1, then try to remove not needed ones.

**Screen-shots:** 
Screenshot_20190330-065845

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by maxbetton**
**Description (required)**

Fixes #2810

Added option to remove a language description when uploading a new photo.

**Tests performed (required)**

Tested betaDebug on Google Pixel 2 XL with API level 28.

**Screenshots (for UI changes only)**

Capture d’écran 2021-01-28 à 14 21 15
Capture d’écran 2021-01-28 à 14 21 46

---

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java [hunk 1]
```diff
@@ -21,195 +22,212 @@
...
-    private final String savedLanguageValue;
+  private HashMap<AdapterView, String> selectedLanguages;
+  private final String savedLanguageValue;
 
-    public UploadMediaDetailAdapter(String savedLanguageValue) {
-        uploadMediaDetails = new ArrayList<>();
-        selectedLanguages = new HashMap<>();
-        this.savedLanguageValue = savedLanguageValue;
-    }
+  public UploadMediaDetailAdapter(String savedLanguageValue) {
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java:35]: These lines shouldn't be edited for this PR. Kindly please revert them.

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Hey @maxbetton , thanks a lot for the PR. It works as intended, I have tested. The only issue is some lines of code are edited while it is not necessary (ie. using double space instead of tabs). After you revert those changes, this PR can be merged. Works as intended :)

File: app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java (hunk 1 above)
Comment (maxbetton) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java:35]: thanks for your review. I auto indent with android studio and that the result. Should i modify anyway ?

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java:35]: Yes, please :/ Otherwise it wont be similar with other files.

**Comment by neslihanturan** [comment]
Hey @maxbetton , any luck with reverting irrelevant changes?:)

**Comment by neslihanturan** [comment]
A kindly reminder @maxbetton :)

**Comment by neslihanturan** [comment]
Closing this one since it seems like a sleeping PR, however it works. Anyone interested to solve the issue can implement the same code. Thanks a lot @maxbetton :)

**Pull request closed without merging**